be/src/util/brpc_stub_cache.h

```diff
@@ -72,18 +73,52 @@ class BrpcStubCache {
 
     doris::PBackendService_Stub* get_stub(const TNetworkAddress& taddr) {
         butil::EndPoint endpoint;
-        if (str2endpoint(taddr.hostname.c_str(), taddr.port, &endpoint)) {
-            LOG(WARNING) << "unknown endpoint, hostname=" << taddr.hostname;
-            return nullptr;
+        if (config::use_fqdn) {
+            std::vector<std::string> addresses;
+            Status status = hostname_to_ip_addrs(taddr.hostname.c_str(), &addresses);
+            if (!status.ok()) {
+                LOG(WARNING) << "failed to get ip from networkAddress";
+                return nullptr;
+            }
+            if (addresses.size() != 1) {
+                LOG(WARNING) << "the number of addresses can only be equal to 1, failed to get ip from networkAddress";
+                return nullptr;
+            }
```

Comment:
Why transfer from FQDN to IP again? How about just using FQDN:port? It should be a common usage on network communication.